Wikipedia/Code/HistoryViewController.swift

```diff
@@ -1,306 +1,377 @@
+
+        let shareArticleAction: WMFHistoryViewModel.ShareRecordAction = { [weak self] frame, historyItem in
+            guard let self else { return }
+            self.share(item:historyItem, frame: frame)
         }
+
+        let onTapArticleAction: WMFHistoryViewModel.OnRecordTapAction = { [weak self] historyItem in
+            guard let self else { return }
+            self.tappedArticle(historyItem)
+        }
+
+        viewModel.onTapArticle = onTapArticleAction
+        viewModel.shareRecordAction = shareArticleAction
+    }
+
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
+        addComponent(hostingController, pinToEdges: true, respectSafeArea: true)
+
+        viewModel.$isEmpty
+            .receive(on: RunLoop.main)
+            .sink { [weak self] isEmpty in
+                self?.deleteButton?.isEnabled = !isEmpty
+            }
+            .store(in: &viewModelCancellables)
     }
 
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        configureNavigationBar()
+    }
+
+    // MARK: - Methods
+
     private func configureNavigationBar() {
-        guard let language  = dataStore.languageLinkController.appLanguage?.languageCode else { return }
-        let wmfLanguage = WMFLanguage(languageCode: language, languageVariantCode: nil)
-        let project = WMFProject.wikipedia(wmfLanguage)
-        let experimentAssignment = (try? WMFActivityTabExperimentsDataController.shared?.getActivityTabExperimentAssignment(project: project)) ?? .control
-        let alignment: WMFNavigationBarTitleConfig.Alignment = experimentAssignment == .control ? .leadingCompact : .centerCompact
-        
-        var titleConfig: WMFNavigationBarTitleConfig = WMFNavigationBarTitleConfig(title: CommonStrings.historyTabTitle, customView: nil, alignment: alignment)
+        let deleteAllButtonText = WMFLocalizedString("history-clear-all", value: "Clear", comment: "Text of the button shown at the top of history which deletes all history {{Identical|Clear}}")
+        var profileButtonConfig: WMFNavigationBarProfileButtonConfig? = nil
+        var tabsButtonConfig: WMFNavigationBarTabsButtonConfig? = nil
+        deleteButton = UIBarButtonItem(title: deleteAllButtonText, style: .plain, target: self, action: #selector(deleteButtonPressed(_:)))
+        deleteButton?.isEnabled = !viewModel.isEmpty
+        let hideNavigationBarOnScroll = !viewModel.isEmpty
+
+        let titleConfig: WMFNavigationBarTitleConfig = WMFNavigationBarTitleConfig(title: CommonStrings.historyTabTitle, customView: nil, alignment: .leadingCompact)
         extendedLayoutIncludesOpaqueBars = false
+
         if #available(iOS 18, *) {
             if UIDevice.current.userInterfaceIdiom == .pad && traitCollection.horizontalSizeClass == .regular {
-                titleConfig = WMFNavigationBarTitleConfig(title: CommonStrings.historyTabTitle, customView: nil, alignment: alignment)
```

Comment:
I think this might be an existing bug (maybe introduced with activity tab?), but we should have the history title here for iPadOS 18 configured to .leadingLarge, so that it matches the other tabs:

leadingCompact:
![IMG_1329](https://github.com/user-attachments/assets/2cd1e1eb-ddb6-411e-acb3-972fbed5d390)

leadingLarge:
![IMG_1330](https://github.com/user-attachments/assets/4608b1a4-1576-47f9-a9a4-7913e3549601)